crates/iceberg/src/spec/table_metadata.rs

```diff
@@ -727,14 +746,10 @@ pub(super) mod _serde {
                     .collect(),
                 default_spec_id: v.default_spec_id,
                 last_partition_id: v.last_partition_id,
-                properties: if v.properties.is_empty() {
-                    None
-                } else {
-                    Some(v.properties)
-                },
+                properties: Some(v.properties),
                 current_snapshot_id: v.current_snapshot_id.or(Some(-1)),
                 snapshots: if v.snapshots.is_empty() {
-                    None
+                    Some(vec![])
```

Comment:
We should not let `properties` and `snapshot` be `Option` if they must be serialized.

Reply:
This is interesting, the spec says `properties` and `snapshot` are both optional, and [this example](https://github.com/apache/iceberg/blob/3d929d459ab21efe313f42754b58b4392d547aee/core/src/test/resources/TableMetadataV2ValidMinimal.json) says it doesn't require these fields.